wasm/sdk.go

```diff
@@ -0,0 +1,800 @@
+	return promiseConstructor.New(handler)
+}
+
+// //
+// // storage SC configurations and blobbers
+// //
+
+func ZBOXGetStorageSCConfig(this js.Value, p []js.Value) interface{} {
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+
+		go func() {
+			result, err := sdk.GetStorageSCConfig()
+
+			if err != nil {
+				// fmt.Println("get_public_encryption_key_failed: " + err.Error())
+				reject.Invoke(js.ValueOf("get_blobber_failed: " + err.Error()))
+				return
+			}
+
+			responseConstructor := js.Global().Get("Response")
+			response := responseConstructor.New(js.ValueOf(result))
+
+			// Resolve the Promise
+			resolve.Invoke(response)
+		}()
+
+		return nil
+	})
+
+	// Create and return the Promise object
+	promiseConstructor := js.Global().Get("Promise")
+	return promiseConstructor.New(handler)
+}
+
+func ZBOXGetBlobbers(this js.Value, p []js.Value) interface{} {
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+
+		go func() {
+			result, err := sdk.GetBlobbers()
+
+			if err != nil {
+				// fmt.Println("get_public_encryption_key_failed: " + err.Error())
+				reject.Invoke(js.ValueOf("get_blobber_failed: " + err.Error()))
+				return
+			}
+
+			responseConstructor := js.Global().Get("Response")
+			response := responseConstructor.New(js.ValueOf(result))
+
+			// Resolve the Promise
+			resolve.Invoke(response)
+		}()
+
+		return nil
+	})
+
+	// Create and return the Promise object
+	promiseConstructor := js.Global().Get("Promise")
+	return promiseConstructor.New(handler)
+}
+
+// // GetBlobber instance.
+func ZBOXGetBlobber(this js.Value, p []js.Value) interface{} {
+	blobberID := p[0].String()
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+
+		go func() {
+			result, err := sdk.GetBlobber(blobberID)
+
+			if err != nil {
+				// fmt.Println("get_public_encryption_key_failed: " + err.Error())
+				reject.Invoke(js.ValueOf("get_blobber_failed: " + err.Error()))
+				return
+			}
+
+			responseConstructor := js.Global().Get("Response")
+			response := responseConstructor.New(js.ValueOf(result))
+
+			// Resolve the Promise
+			resolve.Invoke(response)
+		}()
+
+		return nil
+	})
+
+	// Create and return the Promise object
+	promiseConstructor := js.Global().Get("Promise")
+	return promiseConstructor.New(handler)
+}
+
+// //
+// // ---
+// //
+
+// Ported from `code/go/0proxy.io/zproxycore/handler/wallet.go`
+// Promise code taken from:
+// https://withblue.ink/2020/10/03/go-webassembly-http-requests-and-promises.html
+func GetClientEncryptedPublicKey(this js.Value, p []js.Value) interface{} {
+	clientJSON := p[0].String()
+	chainJSON := p[1].String()
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+
+		go func() {
+			err := initSDK(clientJSON, chainJSON)
```

Comment:
That will cause an issue, as we Init SDK outside of this function

Reply:
Should we just remove all of he `initSDK` inside the SDK function? Assuming we should run `initSDK` before running all of the function? IF yes, then I should also remove the `initSDK` from the `zboxcore/sdk/sdk.go` .